### PR TITLE
[Feature/extensions] Modified Environment settings support 

### PR DESCRIPTION
**Pull request by joshpalis**
### Description
Environment Settings support for extensibility will now pass the entire `Settings` object from OpenSearch node environment to the SDK.

Companion PR : https://github.com/opensearch-project/opensearch-sdk-java/pull/179

### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk-java/issues/78

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/env/EnvironmentSettingsResponse.java [hunk 1]
```diff
@@ -28,59 +22,41 @@
...
-            Object componentSettingValue = componentSetting.get(environmentSettings);
-            componentSettingValues.put(componentSetting, componentSettingValue);
-        }
-        this.componentSettingValues = componentSettingValues;
+    public EnvironmentSettingsResponse(Settings environmentSettings) {
+        this.environmentSettings = environmentSettings;
     }
 
     public EnvironmentSettingsResponse(StreamInput in) throws IOException {
         super(in);
```
Comment (dbwiddis) [inline server/src/main/java/org/opensearch/env/EnvironmentSettingsResponse.java:32]: Nit: I don't like seeing `super(in)` (which in the Response case is a no-op) without a corresponding `out.writeTo()`.  Makes me think too much. :)

**Review by dbwiddis — APPROVED** [review]

File: server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java [hunk 2]
```diff
@@ -477,54 +485,11 @@ public void testHandleActionListenerOnFailureRequest() throws Exception {
...
         // Test EnvironmentSettingsResponse arg constructor
-        EnvironmentSettingsResponse environmentSettingsResponse = new EnvironmentSettingsResponse(settings, componentSettings);
-        assertEquals(componentSettings.size(), environmentSettingsResponse.getComponentSettingValues().size());
-
-        List<Setting<?>> responseSettings = new ArrayList<>();
-        responseSettings.addAll(environmentSettingsResponse.getComponentSettingValues().keySet());
-        assertTrue(responseSettings.containsAll(componentSettings));
-        assertTrue(componentSettings.containsAll(responseSettings));
+        EnvironmentSettingsResponse environmentSettingsResponse = new EnvironmentSettingsResponse(settings);
+        assertEquals(settings, environmentSettingsResponse.getEnvironmentSettings());
```
Comment (owaiskazi19) [inline server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java:492]: We should add a test to compare if the setting present in the list matches the one we have added to the list

Comment (joshpalis) [reply server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java:492]: With this modification to Environment settings support, we are no longer transporting a list of `Setting<T>` objects but rather an entire `Settings` object, so we are not adding anything to any list. As for the comparison test for this transported `Settings` object, this is handled on line 492.

**Comment by joshpalis** [comment]
Summary of EnvironmentSettings workflow modification for future reference :

Previous Design :

Extensions will send OpenSearch a list of Setting objects that it would want to retrieve values from. The following workflow will achieve this support :

1.  Pass in environment settings from Node.java to ExtensionsOrchestrator during node initialization
2.  During createComponents, for each component, the Extension will send a request to OpenSearch containing a List of Setting<T> objects
3.  OpenSearch will receive this list and invoke Setting<T>.get(environmentSettings) to retrieve the corresponding values (or default value if the setting object is not yet registered) and map them to the setting
4.  OpenSearch will respond with the Map<Setting<T> setting, Object value>
5.  The extension will then invoke Map.get(Setting<T>) to retrieve the value of the setting from the OpenSearch environment and then initialize the corresponding component field with this value


New Design : 

While performing create-component integration for Anomaly Detection, it was determined that the entire environment `Settings` is necessary to transport to the extension in order to initialize the `JvmService` object [here](https://github.com/joshpalis/anomaly-detection/blob/572fd8d4f08691326d92535f88f7d5f06ea4ea63/src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java#L271) which is then further used to initialize the `ADCircuitBreakerService`, `MemoryTracker`, `ModelManager `, and `HashRing` objects. These objects are then used to create the `ADTaskManager` component. Since we must transport this object in its entirety, this makes sending List<Setting<T>> objects to OpenSearch for environment settings values obsolete. Thus the following workflow has been modified as such :

1. Pass in environment settings from Node.java to ExtensionsOrchestrator during node initialization
2.  During createComponents, a single `sendEnvironmentSettingsRequest` will be sent to OpenSearch to retrieve the `Settings` object from ExtensionsOrchestrator. 
3. `Settings` object will then be passed to the create-components method of Extension.java, which it will be further passed to any of the objects / components that need it.
4. Components will then use the `Settings` object to determine the registered or default value of the Setting<T> object associated with the component field. For example : https://github.com/joshpalis/anomaly-detection/blob/572fd8d4f08691326d92535f88f7d5f06ea4ea63/src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java#L155

**Review by owaiskazi19 — APPROVED** [review]
Thanks for addressing the comments!

**Pull request merged**